udp-reassembly-offload.md

```diff
@@ -29,4 +29,80 @@ The NDIS interface for URO is used for communication between TCPIP and the NDIS
 
 > **TODO**
 
+ntddndis.h
+```
+#if (NDIS_SUPPORT_NDIS690)
+//
+// values used in UDP receive offload
+//
+#define NDIS_OFFLOAD_PARAMETERS_URO_DISABLED            1
+#define NDIS_OFFLOAD_PARAMETERS_URO_ENABLED             2
+#endif // (NDIS_SUPPORT_NDIS690)
+
+...
+
+#if (NDIS_SUPPORT_NDIS690)
+    struct
+    {
+        UCHAR               IPv4;
+        UCHAR               IPv6;
+    } UdpReceiveOffload;
+#endif // (NDIS_SUPPORT_NDIS690)
+} NDIS_OFFLOAD_PARAMETERS, *PNDIS_OFFLOAD_PARAMETERS;
+
+...
+
+#if (NDIS_SUPPORT_NDIS690)
+typedef struct _NDIS_UDP_RECV_OFFLOAD
+{
+    struct
+    {
+        BOOLEAN Enabled;
+    } IPv4;
+    struct
+    {
+        BOOLEAN Enabled;
+    } IPv6;
+} NDIS_UDP_RECV_OFFLOAD, *PNDIS_UDP_RECV_OFFLOAD;
+#endif
+...
+
+#if (NDIS_SUPPORT_NDIS690)
+  //
+  // UDP Receive Offload
+  //
+  NDIS_UDP_RECV_OFFLOAD        UdpRecvOffload;
+#endif
+
+} NDIS_OFFLOAD, *PNDIS_OFFLOAD;
+```
+
+nbluro.w
+```
+
+
+#if NDIS_SUPPORT_NDIS690
+
+//
+// Per-NetBufferList information for UdpRecvSegCoalesceOffloadInfo.
+//
+typedef struct _NDIS_UDP_RSC_OFFLOAD_NET_BUFFER_LIST_INFO
+{
+    union
+    {
+        struct
+        {
+            ULONG SegCount: 16;
+            ULONG SegSize: 16;
+            ULONG IPVersion: 1;
```

Comment:
what is this field for?

Reply:
I thought it would be useful because URO could be enabled for IPv4 and disabled for IPv6, but on further thinking, I don't think it's actually useful. Will remove.